CommandLine/testing/SimpleTests/data_structure_test.sog/create.edl

```diff
@@ -24,6 +24,22 @@ gtest_assert_eq(ds_list_find_value(list_str, 0), "alarm");
 gtest_assert_eq(ds_list_find_value(list_str, 1), "bank");
 gtest_assert_eq(ds_list_find_value(list_str, 2), "cash");
 
+// When shuffle is repeated with the same seed, it results
+// in a repeated set of orderings modulo 2.
+for (var i = 0; i < 5; ++i) {
+	random_set_seed(69378);
+	ds_list_shuffle(list_str);
+	if (i % 2) {
+		gtest_assert_eq(ds_list_find_value(list_str, 0), "alarm");
+		gtest_assert_eq(ds_list_find_value(list_str, 1), "bank");
+		gtest_assert_eq(ds_list_find_value(list_str, 2), "cash");
+	} else {
+		gtest_assert_eq(ds_list_find_value(list_str, 0), "bank");
+		gtest_assert_eq(ds_list_find_value(list_str, 1), "alarm");
+		gtest_assert_eq(ds_list_find_value(list_str, 2), "cash");
+	}
+}
```

Comment:
This is not correct yet, GM8.1 has a superset of random sequences of the list whose size is equal to the list, not one less as I have in this above.

Reply:
So now I updated it to what I'm getting in ENIGMA with that seed and the other random fix to move the last element. However, I don't have GM8 on hand to see if that's the same as what it had.

Reply:
```
cash alarm bank
bank cash alarm
alarm bank cash
cash alarm bank
bank cash alarm
alarm bank cash
```

Reply:
Thanks tkg, so we're pretty damn close now, GM8 is just one ahead of us for some reason.

Reply:
Oddly the following two will produce Sam's results.
```cpp
return enigma_user::random(x - 2);
return enigma_user::random_integer(x - 3);
```

Reply:
This is so messed up, I changed the test to 4 items and TKG says GM8 just kept repeating only 2 permutations of the list. Those permutations were `bank alarm deer cash` and `alarm bank cash deer`. So something really smells about GM8's shuffle function.

```gml
var list_str;
list_str = ds_list_create();

ds_list_add(list_str, "alarm");
ds_list_add(list_str, "bank");
ds_list_add(list_str, "cash");
ds_list_add(list_str, "deer");

for (i = 0; i < 8; i += 1) {
	random_set_seed(69378);
	ds_list_shuffle(list_str);
	show_message(string(ds_list_find_value(list_str, 0)) + " " +
				 string(ds_list_find_value(list_str, 1)) + " " +
				 string(ds_list_find_value(list_str, 2)) + " " +
				 string(ds_list_find_value(list_str, 3)));
}
```

Reply:
Ok I had a secondary tester confirm Sam's results to me in 8.0 and 8.1. I am guessing Overmars wrote his own shuffle function.

```
B A D C 
A B C D 
B A D C 
A B C D 
B A D C 
A B C D 
B A D C 
A B C D
in 8.0
```